qa/rpc-tests/test_framework/zip317.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#
+# zip317.py
+#
+# Utilities for ZIP 317 conventional fee specification, as defined in https://zips.z.cash/zip-0317.
+#
+
+from test_framework.mininode import COIN
+from decimal import Decimal
+
+# The fee per logical action, in ZAT. See https://zips.z.cash/zip-0317#fee-calculation.
+MARGINAL_FEE = 5000
+
+# The lower bound on the number of logical actions in a tx, for purposes of fee calculation. See
+# https://zips.z.cash/zip-0317#fee-calculation.
+GRACE_ACTIONS = 2
+
+# The Zcashd RPC sentinel value to indicate the conventional_fee when a positional argument is
+# required.
+ZIP_317_FEE = -1
```

Comment:
This is Python, so you don't have to use a number here.
```suggestion
ZIP_317_FEE = None
```
That way, trying to use the sentinel value with arithmetic operations will not silently do the wrong thing:
```python
>>> None + 42
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

Reply:
This only works if we also wrap every single method that takes a fee, internally convert this to `-1` for JSON-RPC, and only ever use the wrapper.

Reply:
OK, good point. I withdraw the suggestion.